File: Data Structures & Algorithms/add-two-numbers/submission-1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* dummy = new ListNode();
        ListNode* temp = dummy;
        int carry = 0;
        while(l1 and l2){
            int tot_sum = l1->val + l2->val+ carry;
            int sum = tot_sum%10;
            carry = tot_sum/10;
            l1->val = sum;
            temp -> next = l1;
            l1 = l1->next;
            l2 = l2->next;
            temp = temp->next;
        }
        while(l1){
            int tot_sum = l1->val+carry;
            int sum = tot_sum%10;
            carry = tot_sum/10;
            l1->val = sum;
            temp -> next = l1;
            l1 = l1->next;
            temp = temp->next;
        }

        while(l2){
            int tot_sum = l2->val+carry;
            int sum = tot_sum%10;
            carry = tot_sum/10;
            l2->val = sum;
            temp -> next = l2;
            l2 = l2->next;
            temp = temp->next;
        }

        if(carry!=0){
            temp->next = new ListNode(carry);
        }

        return dummy->next;

    }
};

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy= ListNode(0)
        temp = head
        nc = 0
        while nc<n:
            temp = temp.next
            nc+=1
        ttemp = dummy
        ttemp.next = head

        while temp!=None:
            ttemp = ttemp.next
            temp = temp.next
        
        ttemp.next = ttemp.next.next
        return dummy.next
